File: axpclient/src/commonMain/kotlin/dk/rohdef/axpclient/shift/WeekPlan.kt
package dk.rohdef.axpclient.shift

internal data class WeekPlan(
    val monday: Weekday,
    val tuesday: Weekday,
    val wednesday: Weekday,
    val thursday: Weekday,
    val friday: Weekday,
    val saturday: Weekday,
    val sunday: Weekday,
) {
    val allDays = listOf(
        monday,
        tuesday,
        wednesday,
        thursday,
        friday,
        saturday,
        sunday,
    )

    val allShifts =
        allDays
            .flatMap { it.allShifts }
}

File: web-app/src/pages/Shifts/WeekPlan.ts
import {Data} from "dataclass";
import {Shift} from "./Shift.ts";

export class WeekPlan extends Data {
    week: string = ""

    monday: Shift[] = []
    tuesday: Shift[] = []
    wednesday: Shift[] = []
    thursday: Shift[] = []
    friday: Shift[] = []
    saturday: Shift[] = []
    sunday: Shift[] = []

    allShifts(): Shift[] {
        return [
            ...this.monday,
            ...this.tuesday,
            ...this.wednesday,
            ...this.thursday,
            ...this.friday,
            ...this.saturday,
            ...this.sunday,
        ]
    }
}
